Userland/Libraries/LibSanitizer/UBSanitizer.cpp

```diff
@@ -46,102 +46,138 @@ static void print_location(const SourceLocation& location)
     }
 }
 
-void __ubsan_handle_load_invalid_value(const InvalidValueData&, ValueHandle) __attribute__((used));
-void __ubsan_handle_load_invalid_value(const InvalidValueData& data, ValueHandle)
+void __ubsan_handle_load_invalid_value(InvalidValueData&, ValueHandle) __attribute__((used));
+void __ubsan_handle_load_invalid_value(InvalidValueData& data, ValueHandle)
 {
+    auto location = data.location.acquire();
+    if (!location.filename())
+        return;
     WARNLN_AND_DBGLN("UBSAN: load-invalid-value: {} ({}-bit)", data.type.name(), data.type.bit_width());
-    print_location(data.location);
+    print_location(location);
 }
 
-void __ubsan_handle_nonnull_arg(const NonnullArgData&) __attribute__((used));
-void __ubsan_handle_nonnull_arg(const NonnullArgData& data)
+void __ubsan_handle_nonnull_arg(NonnullArgData&) __attribute__((used));
+void __ubsan_handle_nonnull_arg(NonnullArgData& data)
 {
+    auto location = data.location.acquire();
+    if (!location.filename())
```

Comment:
Instead of checking `!filename()`can we add a method, like `needs_logging()` or some better name to make the semantics more obvious?

Reply:
good call, added needs_logging() to use instead.